File: dsa/recursion/fib.cpp
#include<iostream>
using namespace std;
int fib(int n){
    if (n==1 | n==0)
    {
        return n;
    }
    
    // 0,1,1,2,3,5,8,13,21
    return fib(n-2) + fib(n-1);
}

int main( )
  {
      cout<< fib(5);

return 0 ;
}

File: python/py/fib.py
def fib(*n):
    if (n==1) :
        return 1
    elif (n==0) :
        return 0
    print(last)
    print(n)
    last = fib(n-1) + fib(n-2)
    


n = int(input("enter th number : "))
print(fib(n-1))
